src/components/swap/PendingModalContent/ProgressIndicator.tsx

```diff
@@ -0,0 +1,189 @@
+
+  const swapConfirmed = swapStatus === TransactionStatus.Confirmed || order?.status === UniswapXOrderStatus.FILLED
+  const wrapConfirmed = useIsTransactionConfirmed(wrapTxHash)
+
+  const swapPending = swapResult !== undefined && !swapConfirmed
+  const wrapPending = wrapTxHash != undefined && !wrapConfirmed
+  const transactionPending = revocationPending || tokenApprovalPending || wrapPending || swapPending
+
+  const [signatureExpiredErrorId, setSignatureExpiredErrorId] = useState('')
+  useEffect(() => {
+    if (swapError instanceof SignatureExpiredError && swapError.id !== signatureExpiredErrorId) {
+      setSignatureExpiredErrorId(swapError.id)
+      onRetryUniswapXSignature?.()
+    }
+  }, [onRetryUniswapXSignature, signatureExpiredErrorId, swapError])
+
+  function getStatus(targetStep: ProgressIndicatorStep) {
+    const currentIndex = steps.indexOf(currentStep)
+    const targetIndex = steps.indexOf(targetStep)
+    if (currentIndex < targetIndex) {
+      return StepStatus.PREVIEW
+    } else if (currentIndex === targetIndex) {
+      return transactionPending ? StepStatus.IN_PROGRESS : StepStatus.ACTIVE
+    } else {
+      return StepStatus.COMPLETE
+    }
+  }
+
+  // TODO: update timeToEnd with dynamic value, calculated as approx. the most recent block confirmation time
+  const stepDetails: Record<ProgressIndicatorStep, StepDetails> = useMemo(
+    () => ({
+      [ConfirmModalState.WRAPPING]: {
+        icon: <CurrencyLogo currency={trade?.inputAmount.currency} />,
+        rippleColor: inputTokenColor,
+        previewTitle: `Wrap ${nativeCurrency.symbol}`,
+        actionRequiredTitle: `Wrap  ${nativeCurrency.symbol} in wallet`,
+        inProgressTitle: `Wrapping  ${nativeCurrency.symbol}...`,
+        timeToEnd: chainId === ChainId.MAINNET ? 20 : 10,
+        delayedTitle: 'Longer than expected...',
+        learnMoreLinkText: `Why do I have to wrap my ${nativeCurrency.symbol}?`,
+        learnMoreLinkHref: SupportArticleURL.WETH_EXPLAINER,
+      },
+      [ConfirmModalState.RESETTING_TOKEN_ALLOWANCE]: {
+        icon: <CurrencyLogo currency={trade?.inputAmount.currency} />,
+        rippleColor: inputTokenColor,
+        previewTitle: `Reset ${trade?.inputAmount.currency.symbol} limit`,
+        actionRequiredTitle: `Reset ${trade?.inputAmount.currency.symbol} limit in wallet`,
+        inProgressTitle: `Resetting ${trade?.inputAmount.currency.symbol} limit...`,
+        timeToEnd: chainId === ChainId.MAINNET ? 20 : 10,
+        delayedTitle: 'Longer than expected...',
+      },
+      [ConfirmModalState.APPROVING_TOKEN]: {
+        icon: <CurrencyLogo currency={trade?.inputAmount.currency} />,
+        rippleColor: inputTokenColor,
+        previewTitle: `Approve ${trade?.inputAmount.currency.symbol} spending`,
+        actionRequiredTitle: `Approve in wallet`,
+        inProgressTitle: 'Approval pending...',
+        timeToEnd: chainId === ChainId.MAINNET ? 20 : 10,
+        delayedTitle: 'Longer than expected...',
```

Comment:
On the approval step, the `Longer than expected` text never appears and you get a blank row
![Screenshot 2023-11-01 at 10 13 01 AM](https://github.com/Uniswap/interface/assets/11512321/1a049cd5-9a0c-4e11-9cc0-3a3fef624f3b)